mars/dataframe/groupby/tests/test_groupby_execution.py

```diff
@@ -580,12 +580,18 @@ def _disallow_reduce(ctx, op):
     ).fetch()
     pd.testing.assert_frame_equal(result.sort_index(), raw.groupby("c2").agg("sum"))
 
-    def _disallow_combine_and_agg(ctx, op):
-        assert op.stage not in (OperandStage.combine, OperandStage.agg)
-        op.execute(ctx, op)
+    rs = np.random.RandomState(0)
+    raw = pd.DataFrame(
+        {
+            "c1": list(range(4)) * 5,
+            "c2": rs.choice(["a", "b", "c"], (20,)),
+            "c3": rs.rand(20),
+        }
+    )
 
+    mdf = md.DataFrame(raw, chunk_size=5)
     r = mdf.groupby("c1").agg("sum")
-    operand_executors = {DataFrameGroupByAgg: _disallow_combine_and_agg}
+    operand_executors = {DataFrameGroupByAgg: _disallow_reduce}
```

Comment:
Why remove the original case that may trigger shuffle groupby?

Reply:
For current strategy, the original case will choose tree method with high probability, I add a new case that will choose shuffle for sure.